platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -1202,6 +1233,29 @@ class HangingProtocolService {
           highestSeriesMatchingScore
         );
 
+        if (imageMatchingRules && imageMatchingRules.length) {
+          const instances = displaySet.images || displaySet.others || [];
+          const filteredInstances = [];
+          for (let instanceIndex = instances.length - 1; instanceIndex >= 0; instanceIndex--) {
+            const instance = instances[instanceIndex];
+            const imagesMatchDetails = this.protocolEngine.findMatch(
+              instance,
+              imageMatchingRules,
+              { instances }
+            );
+
+            if (!imagesMatchDetails.requiredFailed
+              && imagesMatchDetails.details.failed.length === 0) {
+              filteredInstances.push(instance);
+            }
+          }
+
+          DisplaySetService.filterDisplaySetInstances(displaySet.displaySetInstanceUID, filteredInstances);
+          if (instances.length === 0) {
+            return;
+          }
+        }
```

Comment:
Can you also please modify the HangingProtocolService doc with an example on what is `imageMatchingRules` and what can be done with it?

Reply:
sure, I think it is important. added a section to the hpModule docs as well as the DisplayService for the instance level matching and filtering capability.